blade-egui/src/lib.rs

```diff
@@ -81,12 +88,62 @@ impl GuiTexture {
                 subresources: &blade_graphics::TextureSubresources::default(),
             },
         );
-        Self { allocation, view }
+
+        let sampler = context.create_sampler(blade_graphics::SamplerDesc {
+            name,
+            address_modes: {
+                let mode = match options.wrap_mode {
+                    egui::TextureWrapMode::ClampToEdge => blade_graphics::AddressMode::ClampToEdge,
+                    egui::TextureWrapMode::Repeat => blade_graphics::AddressMode::Repeat,
+                    egui::TextureWrapMode::MirroredRepeat => {
+                        blade_graphics::AddressMode::MirrorRepeat
+                    }
+                };
+                [mode; 3]
+            },
+            mag_filter: match options.magnification {
+                egui::TextureFilter::Nearest => blade_graphics::FilterMode::Nearest,
+                egui::TextureFilter::Linear => blade_graphics::FilterMode::Linear,
+            },
+            min_filter: match options.minification {
+                egui::TextureFilter::Nearest => blade_graphics::FilterMode::Nearest,
+                egui::TextureFilter::Linear => blade_graphics::FilterMode::Linear,
+            },
+            mipmap_filter: match options.mipmap_mode {
```

Comment:
this would look like `options.mipmap_mode.map_or(blade_graphics::FilterMode::Linear, my_mapping_function)`